src/NHibernate/Util/EnumerableExtensions.cs

```diff
@@ -6,6 +6,16 @@ namespace NHibernate.Util
 {
 	public static class EnumerableExtensions
 	{
+		public static HashSet<T> ToHashSet<T>(this IEnumerable<T> collection)
```

Comment:
I am sorry but I disagree with the statement that ToHashSet() has to be revered just because it is so simple. For example, ToList() and ToArray() methods from .NET Framework lib do exactly same. It is simply convenient.

Also, if you scroll a bit down you will see ForEach() extension method. Its sole purpose is to avoid writing foreach-statement :) And there are a few more methods like that in different *Extensions.